### DISCUSSION TITLE
Unable to get current page route，Expect a function  import { getCurrentRoute } from 'vue-router'

### DISCUSSION THREAD

**Discussion by alamhubb** [vuejs/router#1901] [Help and Questions] [answered]
### Reproduction

Unable to get current page route

### Steps to reproduce the bug

import { getCurrentRoute } from 'vue-router'

Expect a function from inside the vue-router  getCurrentRoute()

Whether vue-router provides such a function

### Expected behavior

import { getCurrentRoute } from 'vue-router'

Expect a function from inside the vue-router  getCurrentRoute()

Whether vue-router provides such a function

### Actual behavior

vue-router does not provide this function

**Comment by posva** [comment]
Remember to use discussions or the Discord server instead for questions next time 😉

**Reply by alamhubb** [reply]
https://github.com/vuejs/router/issues/1900#issuecomment-1606953108

This discussion has been published for a long time, and there is still no answer. I feel that the discussion efficiency is not as high as issues, so I raised issues

**Comment by posva** [comment, marked as answer]
Use `useRoute()` or `router.currentRoute`

https://router.vuejs.org/api/#Functions-useRoute

**Reply by alamhubb** [reply]
useRoute cannot be used properly in a non-. vue file

**Reply by alamhubb** [reply]
Here's how I do it now

routerUtil.ts
```
import {createRouter, createWebHistory} from "vue-router";


const router = createRouter({
    history: createWebHistory(import.meta.env.BASE_URL),
    routes: [],
})

export function getCurrentRoute(){
    return router.currentRoute
}
```
